Sources/PackageGraph/DependencyResolver.swift

```diff
@@ -747,7 +801,7 @@ public class DependencyResolver<
         //
         // FIXME: We should have a test for this, probably by adding some kind
         // of statistics on the number of backtracks.
-        for constraint in constraints {
+        for constraint in constraints where constraint.requirement != .unversioned {
```

Comment:
Constraint set only holds `VersionSetSpecifier` mapping so it can't tell if there is an mapping of unversioned. I think this loop should instead merge the constraints which of containers which are requested to be resolved, that might make more sense.